File: Python/Unique_Binary_Search_Tree.py
class Solution:
    def numTrees(self, n: int) -> int:
        ans = [0] * (n + 1)
        for i in range(2, n + 1):
            for j in range(1, i + 1):
                ans[i] += ans[j - 1] * ans[i - j]
        return ans[n]

File: Java/uniqueBinarySearchTree.java
class uniqueBinarySearchTree {
    public int numTrees(int n) {
        int[] G = new int[n + 1];
        for (int i; i <= n; i++) {
            for (int j; j <= i; j++) {
                G[i] += G[j - 1] * G[i - j];
            }
        }
        return G[n];
    }
}
